runtime/runtime/src/actions.rs

```diff
@@ -169,10 +168,25 @@ pub(crate) fn execute_function_call(
                 promise_results,
                 apply_state.cache.as_deref(),
             );
-            metrics.report(&apply_state.shard_id.to_string());
+            metrics.report(
+                &apply_state.shard_id.to_string(),
+                &apply_state
+                    .apply_reason
+                    .as_ref()
+                    .map_or_else(|| String::from("unknown"), |r| r.to_string()),
+            );
             r
         }
-        res => res,
+        res => {
+            metrics.report(
```

Comment:
moved this metrics.report call (for when we can find the compiled code in cache even without the contract code) here to avoid double counting cache lookups. please check.

Reply:
We still want to report metrics even if there is a failure in execution, so something smarter needs to be done...

Reply:
Got it. Updated the code to avoid double counting and only invoking the `report_metrics` once. PTAL.